### PR TITLE
Cleanup MultiReleaseJarFile to use newFileSystem behaviors built into JVM

### PR DESCRIPTION
**Pull request by joakime**
Patch against branch `jetty-12.0.x-rework-resources` to show simpler `MultiReleaseJarFile` behavior.

### CODE REVIEW

**Comment by joakime** [comment]
The entire `AnnotationParser` class can be greatly simplified as well.

**Comment by lorban** [comment]
@gregw already started cleaning `AnnotationParser`, and I like this new `MultiReleaseJarFile` a lot.

There are many others cleanups that the `Resource` PR now allows, one of them being the `Resource.__defaultUseCaches` mechanism that uses `URL` to read JAR files.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/FileSystemPool.java [hunk 1]
```diff
@@ -59,7 +58,9 @@ public Resource.Mount mount(URI uri) throws IOException
         {
             try
             {
-                fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap());
+                Map<String, String> env = new HashMap<>();
```
Comment (lorban) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/FileSystemPool.java:61]: This env should be moved to a static field:
```java
private static final Map<String, String> ENV = Map.of("releaseVersion", "runtime");
```

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/MultiReleaseJarFile.java [hunk 2]
```diff
@@ -14,109 +14,59 @@
...
 
-    /**
-     * Construct a multi release jar file
-     *
-     * @param file The file to open
-     * @param javaPlatform The  JVM platform to apply when selecting a version.
-     * @param includeDirectories true if any directory entries should not be ignored
-     * @throws IOException if the jar file cannot be read
-     */
-    public MultiReleaseJarFile(File file, int javaPlatform, boolean includeDirectories) throws IOException
```
Comment (lorban) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/MultiReleaseJarFile.java:63]: Why is the constructor accepting `javaPlatform` gone?

File: jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java [hunk 3]
```diff
@@ -831,18 +833,19 @@ protected void parseJar(Set<? extends Handler> handlers, Resource jarResource) t
                 LOG.debug("Scanning jar {}", jarResource);
 
             MultiException me = new MultiException();
-            // TODO is the jarResource's Path always convertible to File?
-            try (MultiReleaseJarFile jarFile = new MultiReleaseJarFile(jarResource.getPath().toFile(), _javaPlatform, false))
+
+            try (MultiReleaseJarFile jarFile = new MultiReleaseJarFile(jarResource.getPath());
```
Comment (lorban) [inline jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java:837]: Why did you remove the `_javaPlatform`?

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/MultiReleaseJarFile.java (hunk 2 above)
Comment (joakime) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/MultiReleaseJarFile.java:63]: It wasn't used.

File: jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java (hunk 3 above)
Comment (joakime) [reply jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java:837]: Ditto.
It wasn't used.
